src/openvpn/socket.h

```diff
@@ -1113,14 +1113,37 @@ link_socket_extract_tos (struct link_socket *ls, const struct buffer *ipbuf)
 }
 
 /*
+ * Extract TOS bits.  Assumes that ipbuf is a valid IPv6 packet.
+ */
+static inline void
+link_socket_extract_tos_v6 (struct link_socket *ls, const struct buffer *ipbuf)
+{
+  if (ls && ipbuf)
+    {
+      struct openvpn_ipv6hdr *ip6h = (struct openvpn_ipv6hdr *) BPTR (ipbuf);
+      ls->ptos = ((ip6h->version_prio & 0x0f) << 4) | (ip6h->flow_lbl[0] >> 4);
```

Comment:
It's handling a single byte, so it should be... Anyway I'm going to test this on an OpenWrt/LEDE big-endian router too.